src/ucp/core/ucp_request.c

```diff
@@ -308,7 +308,10 @@ static ucp_md_map_t ucp_request_get_invalidation_map(ucp_request_t *req)
         lane      = key->rma_bw_lanes[i];
         cap_flags = ucp_ep_get_iface_attr(ep, lane)->cap.flags;
 
-        if (cap_flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) {
+        if ((cap_flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) &&
```

Comment:
1. seems unrelated fix?
2. maybe use ucp_ep_is_lane_p2p?

Reply:
it is related to get the new gtest passing for TCP

Reply:
will be done in the separate PR when this PR gets merged